scripts/build-release.sh

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script automates the build and testing of the release of a new version
+# of kubefed.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+source "$(dirname "${BASH_SOURCE}")/util.sh"
+ROOT_DIR="$(cd "$(dirname "$0")/.." ; pwd)"
+
+# Arguments
+RELEASE_TAG="${1-}"
+
+# Globals
+QUAY_REPO="${QUAY_REPO:-kubernetes-multicluster/kubefed}"
+GITHUB_REPO="${GITHUB_REPO:-kubernetes-sigs/kubefed}"
+GITHUB_REMOTE_UPSTREAM_NAME="${GITHUB_REMOTE_UPSTREAM_NAME:-upstream}"
+
+function verify-command-installed() {
+  if ! util::command-installed travis; then
+    echo "travis command not found. Please add travis to your PATH and try again." >&2
+    return 1
+  fi
+}
+
+function build-release-artifacts() {
+  ${ROOT_DIR}/scripts/build-release-artifacts.sh ${RELEASE_TAG}
+}
+
+function create-and-push-tag() {
+  # Use the upstream git remote convention name used by hub.
+  # This creates an annotated tag required to ensure that the KubeFed binaries
+  # are versioned correctly.
+  git tag -a "${RELEASE_TAG}" "${GITHUB_REMOTE_UPSTREAM_NAME}/master" -m "Creating release tag ${RELEASE_TAG}"
+  git push "${GITHUB_REMOTE_UPSTREAM_NAME}" "${RELEASE_TAG}"
+}
+
+function travis-build-status() {
+  local status="$(travis branches --repo ${GITHUB_REPO} | grep "${RELEASE_TAG}" | awk '{print $3}')"
+  echo "${status}"
+}
+
+function verify-travis-build-started() {
+  local buildStatus="$(travis-build-status)"
+  [[ "${buildStatus}" == "started" ]]
+}
+
+function verify-travis-build-completed() {
+  local statuses="${passedStatus} failed errored canceled"
+  local buildStatus="$(travis-build-status)"
+
+  for status in ${statuses}; do
+    if [[ "${status}" == "${buildStatus}" ]]; then
+      return 0
+    fi
+  done
+
+  return 1
+}
+
+function verify-continuous-integration() {
+  local passedStatus="passed"
```

Comment:
I find it a bit strange to see this declared as a local but then used in `verify-travis-build-completed`.  Maybe just make it a global given that its a constant anyway?